### ISSUE TITLE
Accuracy decrease when norm_clip decrease

### ISSUE THREAD

**Issue by YiheWang** [tensorflow/privacy#182]
Hi! As I know, the noise_multiplier is the ratio of noise_stddev to sensitivity. And the sensitivity is proportional to l2_clip_norm. So when we fixed the noise_multiplier, noise_stddev will be smaller if the l2_clip_norm is smaller, which leads to a smaller Gaussian Noise. And Smaller Gaussian Noise means better accuracy. However, I am confused about my training result.

I am trying to build a differentially private face classifier. 

Here is the parameter for first network. Epsilon = 22.4 here. The final  val_accuracy is 0.7235
l2_norm_clip = 10
noise_multiplier = 1.0
num_microbatches = 20
learning_rate = 0.002
batch_size = 20
epochs = 400
delta = 0.001

Now I change l2_norm_clip = 5 and keep other parameters. Epsilon is still 22.4 here. The final val_accuracy is 0.6029.

So this result confused me. In my understanding, the noise_stddev is smaller when l2_norm_clip = 5, which means smaller Gaussian Noise. My accuracy should be higher than before. Could someone helps me understand why this situation is here? Do I misunderstand any part? Thank you in advance for your help!

**Comment by galenmandrew** [comment]
You're correct that a smaller clip implies less noise, which on its own would improve performance, but I can think of two ways a small clip could lead to worse performance.

First, a small clip means you are throwing away more information from the magnitude of the gradients. Maybe there are some gradients delivering a strong signal with norms close to 12. With l2_norm_clip = 10 those vectors are mostly preserved. With l2_norm_clip = 5 you are cutting them more than in half and mixing them with the smaller gradients.

Second, aggressive clipping means the averaged gradient will me smaller, which can lead to slower convergence. Imagine your clip is 1e-4... Then the model cannot make any progress because its updates are miniscule. If this is your problem, it could be remedied by increasing the learning_rate.

It is recommended that you use the adaptive clipping strategy outlined in [this paper](https://arxiv.org/abs/1905.03871) and implemented [here](https://github.com/tensorflow/privacy/blob/8850c23f67d31a3baee2490224dea9a605581ddd/tensorflow_privacy/privacy/dp_query/quantile_adaptive_clip_sum_query.py).

**Comment by YiheWang** [comment]
Thank you for your help!
